{"pageProps":{"questions":[{"id":"q5AwKG2bKZUqK43QR4xq","url":"https://www.examtopics.com/discussions/python-institute/view/141964-exam-pcap-31-03-topic-1-question-37-discussion/","answer":"C","answer_description":"","answers_community":["C (100%)"],"choices":{"D":"it outputs [1, 3]","B":"it outputs [4, 2, 0]","C":"it outputs [3, 1]","A":"the code is erroneus and it will not execute"},"topic":"1","answer_ET":"C","question_images":[],"exam_id":193,"answer_images":[],"question_text":"What is the expected behavior of the following code?\nmy_list = [i for i in range(5)]\nm = [my_list[i] for i in range(4, 0, -1) if my_list[i] % 2 != 0] print(m)","isMC":true,"question_id":31,"discussion":[{"poster":"Dave304409","upvote_count":"3","timestamp":"1720016880.0","content":"Selected Answer: C\nC is correct","comment_id":"1241471"},{"poster":"DKAT2023","content":"c is correct","upvote_count":"2","timestamp":"1719852360.0","comment_id":"1240248"},{"comment_id":"1237462","content":"it outputs [3,1]","poster":"b4779c5","timestamp":"1719406200.0","upvote_count":"2"}],"unix_timestamp":1717615380,"timestamp":"2024-06-05 21:23:00"},{"id":"cmwQUwLCoRwLrFXaaHs7","topic":"1","answer_description":"","answers_community":["AC (67%)","C (33%)"],"question_images":["https://img.examtopics.com/pcap-31-03/image22.png"],"answer_images":[],"question_id":32,"choices":{"C":"a != b","A":"a is not None","D":"b() == 4","B":"a() == 4"},"question_text":"Assuming that the following code has been executed successfully, select the expressions which evaluate to True (Choose two.)\n//IMG//","discussion":[{"poster":"Dave304409","comment_id":"1241504","comments":[{"content":"A, C us correct","upvote_count":"1","timestamp":"1720021920.0","comment_id":"1241506","poster":"Dave304409"}],"upvote_count":"1","timestamp":"1720021920.0","content":"Selected Answer: C\nC is correct"},{"content":"A and C are correct","upvote_count":"4","timestamp":"1719852840.0","comment_id":"1240250","poster":"DKAT2023"},{"comment_id":"1221675","upvote_count":"2","content":"Selected Answer: AC\nAs tested","poster":"a4129fd","timestamp":"1717082340.0"}],"answer":"AC","url":"https://www.examtopics.com/discussions/python-institute/view/141612-exam-pcap-31-03-topic-1-question-38-discussion/","unix_timestamp":1717082340,"answer_ET":"AD","isMC":true,"timestamp":"2024-05-30 17:19:00","exam_id":193},{"id":"C37MFtYQ6a2rijZcMF3o","answer_description":"","answers_community":["AD (100%)"],"timestamp":"2024-06-26 15:10:00","isMC":true,"question_images":[],"url":"https://www.examtopics.com/discussions/python-institute/view/142955-exam-pcap-31-03-topic-1-question-39-discussion/","discussion":[{"poster":"jaycosmo_23","upvote_count":"1","timestamp":"1733727000.0","comment_id":"1323900","content":"Selected Answer: AD\nCorrect Answers\nA. If invoking open() fails, an exception is raised\nD. open() is a function which returns an object that represents a physical file"},{"upvote_count":"2","poster":"jarun2024","content":"ANS: A & D","comment_id":"1278389","timestamp":"1725467760.0"},{"content":"A second parameter is not necessary because it uses default parameters. See the documentation:\n\nhttps://docs.python.org/3/library/functions.html#open\n\nIf the mode is not specified, it defaults to 'r'.\n\nTherefore, the correct answers are A and D.","timestamp":"1724083260.0","comment_id":"1268819","poster":"kino_1994","upvote_count":"2"},{"content":"A, C is correct","upvote_count":"2","comment_id":"1241509","timestamp":"1720021980.0","poster":"Dave304409"},{"poster":"DKAT2023","comment_id":"1240322","content":"sorry , A and C are corrects not D,","timestamp":"1719859740.0","upvote_count":"1"},{"timestamp":"1719856020.0","poster":"DKAT2023","upvote_count":"2","content":"A and D are corrects","comment_id":"1240278"},{"timestamp":"1719407400.0","comment_id":"1237468","upvote_count":"1","poster":"b4779c5","content":"the correct anserws are AandD"}],"answer_images":[],"exam_id":193,"unix_timestamp":1719407400,"question_text":"Which of the following statements are true? (Choose two.)","choices":{"A":"if invoking open() fails, an exception is raised","B":"instd, outstd, errstd are the names of pre-opened streams","C":"open() requires a second argument","D":"open() is a function which returns an object that represents a physical file"},"answer":"AD","answer_ET":"AC","question_id":33,"topic":"1"},{"id":"kLxm5T72xJv90n4lit99","url":"https://www.examtopics.com/discussions/python-institute/view/140849-exam-pcap-31-03-topic-1-question-4-discussion/","answer_ET":"BC","answer_description":"","discussion":[{"timestamp":"1716097680.0","content":"Only option C, `random.choice([1,2,3]) > 0` will return TRUE.\nAll other options are FALSE.","poster":"narayanan010","upvote_count":"6","comment_id":"1213618"},{"poster":"Sharif24","comment_id":"1272397","content":"Selected Answer: C\nimport random\nv1 = random.random()\nv2 = random.random()\nprint(v1 ==v2) #False\nprint(v1>1) # False\nprint(random.choice(([1,2,3],1))>0) # True","upvote_count":"1","timestamp":"1724643720.0"},{"timestamp":"1719602100.0","upvote_count":"1","poster":"DKAT2023","content":"C is the correct","comment_id":"1238861"},{"comment_id":"1217634","timestamp":"1716563100.0","content":"Option C is correct \nExplanation:\nimport random \nv1 = random.random() \nv2 = random.random()\nrandom.choice([1,2,3]) > 0` will return TRUE.\n\nAll others options will return output as FALSE.","poster":"Nikhil_Durgesh","upvote_count":"3"}],"timestamp":"2024-05-19 07:48:00","question_id":34,"question_text":"Assuming that the code below has been executed successfully, which of the following expressions will always evaluate to True? (Choose two.) import random v1 = random.random() v2 = random.random()","question_images":[],"exam_id":193,"topic":"1","answer":"C","answer_images":[],"choices":{"C":"random.choice([1,2,3]) > 0","A":"len(random.sample([1,2,3],1)) > 2","B":"v1 == v2","D":"v1>1"},"isMC":true,"unix_timestamp":1716097680,"answers_community":["C (100%)"]},{"id":"t0GhbZZf1lUTsRMuWV9p","isMC":true,"question_images":["https://img.examtopics.com/pcap-31-03/image23.png"],"answer":"B","exam_id":193,"url":"https://www.examtopics.com/discussions/python-institute/view/143144-exam-pcap-31-03-topic-1-question-40-discussion/","timestamp":"2024-07-01 19:56:00","unix_timestamp":1719856560,"answers_community":["B (100%)"],"answer_description":"","question_text":"What is the expected output of the following code if there is no file named non_existing_file inside the working directory?\n//IMG//","question_id":35,"answer_images":[],"discussion":[{"upvote_count":"1","comment_id":"1347757","content":"Selected Answer: B\nThe first number is not \"1\" because the try block raises an exception before it reaches the print function. The last number is \"2\" because of the except block and, additionally, the else block not being executed.","poster":"Horsefeathers","timestamp":"1738045740.0"},{"content":"B is correct","poster":"DKAT2023","timestamp":"1719856560.0","upvote_count":"3","comment_id":"1240285"}],"answer_ET":"B -","choices":{"A":"2 2 3","B":"2 2","D":"1 2 2","C":"1 3"},"topic":"1"}],"exam":{"name":"PCAP-31-03","numberOfQuestions":90,"isBeta":false,"id":193,"isImplemented":true,"lastUpdated":"12 Apr 2025","isMCOnly":true,"provider":"Python-institute"},"currentPage":7},"__N_SSP":true}